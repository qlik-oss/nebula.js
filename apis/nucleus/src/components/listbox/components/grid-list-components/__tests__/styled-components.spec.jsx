import React from 'react';
import { create, act } from 'react-test-renderer';
import getStyledComponents /* , { classes } */ from '../styled-components';

describe('styled-components', () => {
  let renderer;
  let render;

  describe('StyledFixedSizeList', () => {
    beforeAll(async () => {
      const { StyledFixedSizeList } = getStyledComponents();
      render = async () => {
        await act(async () => {
          renderer = create(
            <StyledFixedSizeList itemSize={1} height={100}>
              <div>An item</div>
            </StyledFixedSizeList>
          );
        });
      };
    });

    it('should return a renderable base component for List', async () => {
      await render();
      const rootElement = renderer.root;
      expect(rootElement.props).toMatchSnapshot();
      expect(renderer.toTree()[0].props.cache.registered).toMatchSnapshot();
    });
  });

  describe('StyledFixedSizeGrid', () => {
    beforeAll(async () => {
      const { StyledFixedSizeGrid } = getStyledComponents();
      render = async () => {
        await act(async () => {
          renderer = create(
            <StyledFixedSizeGrid itemSize={1} height={100} width={150} columnWidth={50} rowHeight={10}>
              <div>An item</div>
            </StyledFixedSizeGrid>
          );
        });
      };
    });

    it('should return a renderable base component for Grid', async () => {
      await render();
      const rootElement = renderer.root;
      expect(rootElement.props).toMatchSnapshot();
      expect(renderer.toTree()[0].props.cache.registered).toMatchSnapshot();
    });
  });
});
