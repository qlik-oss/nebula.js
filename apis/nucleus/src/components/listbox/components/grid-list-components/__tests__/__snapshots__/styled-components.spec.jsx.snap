// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled-components StyledFixedSizeGrid should return a renderable base component for Grid 1`] = `
{
  "children": <div>
    An item
  </div>,
  "columnWidth": 50,
  "direction": "ltr",
  "height": 100,
  "itemData": undefined,
  "itemSize": 1,
  "rowHeight": 10,
  "useIsScrolling": false,
  "width": 150,
}
`;

exports[`styled-components StyledFixedSizeGrid should return a renderable base component for Grid 2`] = `
{
  "css-qk0o61": "&.ListBox-styledScrollbars{scrollbar-color:#BBB #f1f1f1;&::-webkit-scrollbar{width:10px;height:10px;}&::-webkit-scrollbar-track{background-color:#f1f1f1;}&::-webkit-scrollbar-thumb{background-color:#BBB;border-radius:1rem;}&::-webkit-scrollbar-thumb:hover{background-color:#555;}}&::-webkit-scrollbar{width:10px;height:10px;}&::-webkit-scrollbar-track{background-color:#f1f1f1;}&::-webkit-scrollbar-thumb{background-color:#BBB;border-radius:1rem;}&::-webkit-scrollbar-thumb:hover{background-color:#555;}",
}
`;

exports[`styled-components StyledFixedSizeList should return a renderable base component for List 1`] = `
{
  "children": <div>
    An item
  </div>,
  "direction": "ltr",
  "height": 100,
  "itemData": undefined,
  "itemSize": 1,
  "layout": "vertical",
  "overscanCount": 2,
  "useIsScrolling": false,
}
`;

exports[`styled-components StyledFixedSizeList should return a renderable base component for List 2`] = `
{
  "css-qk0o61": "&.ListBox-styledScrollbars{scrollbar-color:#BBB #f1f1f1;&::-webkit-scrollbar{width:10px;height:10px;}&::-webkit-scrollbar-track{background-color:#f1f1f1;}&::-webkit-scrollbar-thumb{background-color:#BBB;border-radius:1rem;}&::-webkit-scrollbar-thumb:hover{background-color:#555;}}&::-webkit-scrollbar{width:10px;height:10px;}&::-webkit-scrollbar-track{background-color:#f1f1f1;}&::-webkit-scrollbar-thumb{background-color:#BBB;border-radius:1rem;}&::-webkit-scrollbar-thumb:hover{background-color:#555;}",
}
`;
